docs/processors.md

```diff
@@ -0,0 +1,158 @@
+## Processors
+
+A processor is a component that operates on a single record that flows through a pipeline. It can either change the record
+(i.e. **transform** it) or **filter** it out based on some criteria. Since they are part of pipelines, making yourself 
+familiar with [pipeline semantics](/docs/pipeline_semantics.md) is highly recommended.
+
+![Pipeline](data/pipeline_example.svg)
+
+Processors are **optional** components in a pipeline, i.e. a pipeline can be started without them. They are always attached 
+to a single parent, which can be either a connector or a pipeline. With that, we can say that we have the following types 
+of processors:
+1. **Source processors**: these processors only receive messages originating at a specific source connector. Source 
+processors are created by specifying the corresponding source connector as the parent entity.
+2. **Pipeline processors**: these processors receive all messages that flow through the pipeline, regardless of the
+   source or destination. Pipeline processors are created by specifying the pipeline as the parent entity.
+3. **Destination processors**: these processors receive only messages that are meant to be sent to a specific
+   destination connector. Destination processors are created by specifying the corresponding destination connector as the
+   parent entity.
+
+Given that every processor can have one (and only one) parent, processors cannot be shared. In case the same processing 
+needs to happen for different sources or destinations, you have two options:
+1. If records from all sources (or all destinations) need to be processed in the same way, then you can obviously create
```

Comment:
I would remove "obviously".

Reply:
I like to add a little bit of less formal vocabulary, because otherwise (at least to me), the docs can become quite boring.: ) But I completely see the arguments for having it more formal too.